src/Microsoft.OpenApi.OData.Reader/PathItem/NavigationPropertyPathItemHandler.cs

```diff
@@ -80,16 +83,26 @@ protected override void SetOperations(OpenApiPathItem item)
                 {
                     if (LastSegmentIsKeySegment)
                     {
-                        UpdateRestrictionsType updateEntity = Context.Model.GetRecord<UpdateRestrictionsType>(_entityType);
-                        if (updateEntity?.IsUpdatable ?? true)
+                        UpdateRestrictionsType entityUpdateRestriction = Context.Model.GetRecord<UpdateRestrictionsType>(_navPropEntityType);
+                        UpdateRestrictionsType navPropUpdateRestriction = Context.Model.GetRecord<UpdateRestrictionsType>(NavigationProperty);
+                        if ((entityUpdateRestriction?.IsUpdatable ?? true) ||
+                            (navPropUpdateRestriction?.IsUpdatable ?? true))
                         {
-                            AddUpdateOperation(item, restriction);
+                            UpdateRestrictionsType updateRestrictionType = entityUpdateRestriction ?? navPropUpdateRestriction;
```

Comment:
```suggestion
                            UpdateRestrictionsType updateRestrictionType = navPropUpdateRestriction ?? entityUpdateRestriction;
```

So the nav prop restriction can act as an override to the entity one.

Reply:
If we have this example:
```
<NavigationProperty Name="recoveryKeys" Type="Collection(graph.bitlockerRecoveryKey)" ContainsTarget="true">
          <Annotation Term="Org.OData.Core.V1.Description" String="The recovery keys associated with the bitlocker entity." />
        </NavigationProperty>
```

Where the navigation property `recoveryKeys` and the entity type targeted by the navigation property `bitlockerRecoveryKey` are both annotated with conflicting restrictions annotations, shouldn't the navigation property's entity type restriction annotation take precedence?

Reply:
Do we want navigation restrictions to override the entity type (that's targeted by the navigation property) restrictions? 
I haven't seen any use case yet for this and I am not sure about the precedence.
@darrelmiller ?

Reply:
Precedence is nav property overrides entity type.  A scenario would be "this entity can be deleted", however, "you can't delete the entity from this nav property collection".